pkg/client/remotecommand/remotecommand_test.go

```diff
@@ -62,16 +66,22 @@ func (c *fakeUpgradeConnection) CreateStream(headers http.Header) (httpstream.St
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	stream := &fakeUpgradeStream{}
+	stream := &fakeUpgradeStream{
+		closeChan: make(chan struct{}, 10),
```

Comment:
Why 10? The fact that this isn't 0 makes me pretty sure that your Reset/Close calls are not idempotent. Should they be?